server/user/user.controller.js

```diff
@@ -44,9 +44,32 @@ function login({ body }, res) {
     });
 }
 
+function updateProfile({ user, body: { userCredentials } }, res, next) {
+  const { email, firstName, lastName } = userCredentials;
+  return user.update({ email, firstName, lastName })
+    .then(({ profile }) => res.json({ user: profile }))
+    .catch(err => next(err));
+}
+
+function changePassword({ user, body: { currentPassword, newPassword } }, res) {
+  return user.authenticate(currentPassword)
+    .then(user => user || createError(NOT_FOUND, 'Incorrect current password'))
```

Comment:
Use `BAD_REQUEST` instead with message that does not actually explain reason of failure to prevent potential misuse.